### PR TITLE
feat(lib/runtime): add extra required runtime imports for parachain validation

### PR DESCRIPTION
**Pull request by kishansagathiya**
This PR adds following import calls:
- ext_default_child_storage_clear_prefix_version_2
- ext_default_child_storage_root_version_2
- ext_offchain_index_clear_version_1

## Tests

```sh
go test -tags integration github.com/ChainSafe/gossamer
```

## Issues

#2989

## Primary Reviewer

@timwu20

### CODE REVIEW

**Review by kanishkatn — APPROVED** [review]
looks good!

**Comment by kishansagathiya** [comment]
A corresping PR in polkadot spec repo for tests to work https://github.com/ChainSafe/polkadot-spec/pull/15

File: lib/runtime/wasmer/imports_test.go [hunk 1]
```diff
@@ -1283,6 +1301,54 @@ func Test_ext_default_child_storage_clear_version_1(t *testing.T) {
 	require.Nil(t, val)
 }
 
+func Test_ext_default_child_storage_clear_prefix_version_2(t *testing.T) {
```
Comment (EclesioMeloJunior) [inline lib/runtime/wasmer/imports_test.go:1304]: Could you include a test when the limit is not 0?

File: lib/runtime/wasmer/imports.go [hunk 2]
```diff
@@ -1060,6 +1063,33 @@ func ext_default_child_storage_clear_prefix_version_1(context unsafe.Pointer, ch
...
+		logger.Errorf("failed to decode limit: %s", err)
+	}
+
+	err = storage.ClearPrefixInChildWithLimit(keyToChild, prefix, *limit)
+	if err != nil {
+		logger.Errorf("failed to clear prefix in child with limit: %s", err)
+	}
+
+	// TODO: Should this always be 0 or could this be something else as well?
+	return C.int64_t(0)
```
Comment (EclesioMeloJunior) [inline lib/runtime/wasmer/imports.go:1090]: Following the spec the returned value should be a pointer-size to the following variant:

- Variant 0 indicates that all keys of the child storage have been removed, followed by the number of removed keys
- Variant 1 indicates that there are remaining keys, followed by the number of removed keys.

ref: https://spec.polkadot.network/chap-host-api#id-version-2-prototype-3

You can see how they implemented it in substrate:

```rs
#[derive(PassByCodec, Encode, Decode)]
pub enum KillStorageResult {
	/// All keys to remove were removed, return number of iterations performed during the
	/// operation.
	AllRemoved(u32),
	/// Not all key to remove were removed, return number of iterations performed during the
	/// operation.
	SomeRemaining(u32),
}

#[version(2)]
fn clear_prefix(&mut self, prefix: &[u8], limit: Option<u32>) -> KillStorageResult
```
- clear prefix v2: https://github.com/paritytech/substrate/blob/d58481569817f4a2074dbab014fa629dea7e0ec9/primitives/io/src/lib.rs#L186
- the returned variant: https://github.com/paritytech/substrate/blob/d58481569817f4a2074dbab014fa629dea7e0ec9/primitives/io/src/lib.rs#L96

**Review by EclesioMeloJunior — CHANGES_REQUESTED** [review]

File: lib/runtime/storage/trie.go [hunk 3]
```diff
@@ -259,6 +259,21 @@ func (s *TrieState) ClearPrefixInChild(keyToChild, prefix []byte) error {
...
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	child, err := s.t.GetChild(keyToChild)
+	if err != nil {
+		return 0, false, err
+	}
+	if child == nil {
+		return 0, false, nil
+	}
```
Comment (EclesioMeloJunior) [inline lib/runtime/storage/trie.go:272]:
```suggestion
	if err != nil || child == nil {
		return 0, false, err
	}
```

**Review by EclesioMeloJunior — APPROVED** [review]
LGTM, just a small suggestion

**Pull request merged**